### PR TITLE
Parametrize bucket creation test cases

### PR DESCRIPTION
**Pull request by waynesun09**
Update test function with extra parameter amount and interface.
Parametrize tests with current sets and three new sets.

Signed-off-by: Wayne Sun <gsun@redhat.com>

### CODE REVIEW

File: tests/manage/mcg/test_bucket_creation.py [hunk 1]
```diff
@@ -12,40 +12,41 @@
...
+            *[3, 'S3'],
+            marks=[pytest.mark.polarion_id("OCS-1298"), tier1, acceptance]
+        ),
+        pytest.param(
+            *[3, 'CLI'], marks=[tier1, acceptance, noobaa_cli_required]
+        ),
+        pytest.param(
+            *[3, 'OC'], marks=[tier1, acceptance]
+        ),
+        pytest.param(
```
Comment (RazTamir) [inline tests/manage/mcg/test_bucket_creation.py]: Is this a new test case?
Polarion_id?

Comment (MeridianExplorer) [reply tests/manage/mcg/test_bucket_creation.py]: I wouldn't consider it a new test case - it's just a more varied and thorough parametrization of the same test. I don't think a separate Polarion TC is needed

**Comment by MeridianExplorer** [comment]
Great PR - no comments whatsoever.

One thing regarding amounts - I think it'd be good to do 3, 100, and 1,000 on each API. Besides that, approved.

@ebenahar - Once we begin working with large amounts (like 1,000), I worry about the possibility of bucketname collisions (`create_unique_resource_name` *is* finite and not truly fully random). We need to think about this.

**Comment by petr-balogh** [comment]
@MeridianExplorer I think we cannot think about numbers like 1000 buckets as the maximum limit we can request in AWS is 1000:
```
A bucket is owned by the AWS account that created it. By default, you can create up to 100 buckets in each of your AWS accounts. If you need additional buckets, you can increase your account bucket limit to a maximum of 1,000 buckets by submitting a service limit increase. For information about how to increase your bucket limit, see AWS Service Limits in the AWS General Reference.
```
Took this from: https://docs.aws.amazon.com/AmazonS3/latest/dev/BucketRestrictions.html

Our current limit is 200 I think so no test can crate at the moment that amount of s3 buckets otherwise we will hit the limits very soon. If we need more than 200 we need to open support case for increase. Just let me know what limit we need and I will open the request if needed @ebenahar .

**Comment by MeridianExplorer** [comment]
@petr-balogh I think there's a slight confusion here -
At any given stage, `test_bucket_creation` does not alter any AWS state. All bucket creation and deletion ops are performed in front of NooBaa.
As part of the deployment process on AWS, NooBaa creates a bucket on AWS S3. This bucket is then referred to as the 'Cloud resource', 'Backing store' or 'Target bucket'. What it means practically is that NooBaa will be able to use it as an underlying foundation for *its own* buckets. Whenever we create a bucket in front of the NooBaa S3 endpoint, it will create what I personally call _NooBucket_. It's a virtual bucket, managed by NooBaa, and stored on the target bucket in AWS. This allows you to create millions of buckets, if you wish, but all will be stored in the same AWS bucket.

Thus, this test actually only needs one AWS bucket - the default one that's created in the deployment process.

**Comment by petr-balogh** [comment]
> @petr-balogh I think there's a slight confusion here -
> At any given stage, `test_bucket_creation` does not alter any AWS state. All bucket creation and deletion ops are performed in front of NooBaa.
> As part of the deployment process on AWS, NooBaa creates a bucket on AWS S3. This bucket is then referred to as the 'Cloud resource', 'Backing store' or 'Target bucket'. What it means practically is that NooBaa will be able to use it as an underlying foundation for _its own_ buckets. Whenever we create a bucket in front of the NooBaa S3 endpoint, it will create what I personally call _NooBucket_. It's a virtual bucket, managed by NooBaa, and stored on the target bucket in AWS. This allows you to create millions of buckets, if you wish, but all will be stored in the same AWS bucket.
> 
> Thus, this test actually only needs one AWS bucket - the default one that's created in the deployment process.

Ok, thanks for the explanation. I didn't take a look at bucket_factory what it actually does so was confused that there are created buckets directly in AWS instead of via noobaa. Than it should be ok and my concerns go away. Thanks

File: tests/manage/mcg/test_bucket_creation.py (hunk 1 above)
Comment (ebenahar) [reply tests/manage/mcg/test_bucket_creation.py]: polarion_id seems to be deleted below.
@MeridianExplorer I suggest we create separate polarion test cases for the different amounts. The different interfaces don't need separate test cases

Comment (MeridianExplorer) [reply tests/manage/mcg/test_bucket_creation.py]: @waynesun09 - Can you please create the appropriate Polarion test cases and update them in the code?

File: tests/manage/mcg/test_bucket_creation.py [hunk 2]
```diff
@@ -20,19 +20,34 @@
             marks=[pytest.mark.polarion_id("OCS-1298"), tier1, acceptance]
         ),
         pytest.param(
-            *[3, 'CLI'], marks=[tier1, acceptance, noobaa_cli_required]
+            *[3, 'CLI'],
+            marks=[tier1, acceptance, noobaa_cli_required,
+                   pytest.mark.polarion_id("OCS-1377")]
```
[the remark this answers is not in the document]
Comment (MeridianExplorer) [reply tests/manage/mcg/test_bucket_creation.py]: @waynesun09 - this is actually an incorrect ID - BackingStores are a separate concept that are implemented only as part of the multi-region test (which will be up for PR in the upcoming days).
The correct ID in this case is 1298.

File: tests/manage/mcg/test_bucket_creation.py [hunk 3]
```diff
@@ -20,19 +20,34 @@
...
         ),
         pytest.param(
             *[3, 'OC'], marks=[tier1, acceptance]
```
[the remark this answers is not in the document]
Comment (MeridianExplorer) [reply tests/manage/mcg/test_bucket_creation.py]: @waynesun09 - right now, 1298 includes creation with all three APIs. Thus, all the basic tests should refer to 1298.

File: tests/manage/mcg/test_bucket_creation.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (waynesun09) [reply tests/manage/mcg/test_bucket_creation.py]: Updated

File: tests/manage/mcg/test_bucket_creation.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (waynesun09) [reply tests/manage/mcg/test_bucket_creation.py]: Updated

**Pull request merged**